### PR TITLE
refactor(emotion): applying postcss-bidirection functionality

### PR DESCRIPTION
**Pull request by ke1k0**
Closes: INSTUI-2735

Since postcss is not working with Emotion, we needed to replace the functionality of
postcss-bidirection plugin with a local polyfill.
TEST PLAN:
Making sure ltr and rtl styles generate perfectly.

### CODE REVIEW

**Review by HerrTopi — APPROVED** [review]

File: packages/emotion/src/styleUtils/index.js [hunk 1]
```diff
@@ -107,26 +115,38 @@ const withStyle = decorator((ComposedComponent, generateStyle) => {
...
   hoistNonReactStatics(WithStyle, ComposedComponent)
   WithStyle.displayName = `WithStyle(${ComposedComponent.displayName})`
 
   return WithStyle
 })
+
+/**
+ * Applies necessary polyfills on the style object
+ * @param {object} styleObject
+ * @returns {object} the modified style object
```
Comment (matyasf) [inline packages/emotion/src/styleUtils/index.js:145]: This returns sometimes `undefined` (bidirectional.js line 152), if this is expected it should be mentioned here.

File: packages/emotion/src/styleUtils/index.js [hunk 2]
```diff
@@ -82,7 +83,14 @@ const useTheme = () => {
...
 
-  return makeStyles(theme, componentName, generateStyle, props)(...rest)
+  const styles = calculateStyles(
+    theme,
+    componentName,
+    generateStyle,
+    props
+  )(...rest)
+
+  return polyfillStyles(styles)
```
Comment (matyasf) [inline packages/emotion/src/styleUtils/index.js:93]: This returns sometimes undefined (bidirectional.js line 152), if this is expected it should be mentioned in the description.

File: packages/emotion/src/styleUtils/index.js (hunk 1 above)
Comment (ke1k0) [reply packages/emotion/src/styleUtils/index.js:153]:
```js
function bidirectionalPolyfill(styles) {
  const stylesCopy = {}
  //..... rest of code, forEach ....
  return stylesCopy
```
It is an empty object by default, so it always gives back an object. In the forEach loop it just skips properties with no value.

File: packages/emotion/src/styleUtils/index.js (hunk 2 above)
Comment (ke1k0) [reply packages/emotion/src/styleUtils/index.js:79]: answered above

File: packages/emotion/src/styleUtils/index.js (hunk 1 above)
Comment (matyasf) [reply packages/emotion/src/styleUtils/index.js:153]: oh yeah, I missed that part, it looks good :)

**Review by matyasf — APPROVED** [review]

**Comment by ke1k0** [comment]
Done in another PR: https://github.com/instructure/instructure-ui/pull/318

**Pull request closed without merging**